### DISCUSSION TITLE
Hide startup vulcan error banner

### DISCUSSION THREAD

**Discussion by telanus** [lutris/lutris#4905] [Ideas]
Is it possible to hide the nag screen when Lutris don't find vulcan on a system. AFAIK there used to be a checkmark to hide it after the first time it comes up, but seems it disappeared in the last week

**Comment by danieljohnson2** [comment]
I can't find the issue #, but @strycore seemed to be quite exasperated at 'defects' that were really just that the user's drivers were not set up properly, and has just now removed that 'don't tell me again' dialog altogether.

But I do agree that nagging the user for not having Vulkan on every startup is not nice.

I assume I would not be welcome to just restore that checkboxy dialog and use it; I have already added other warnings in the configuration window that can't be suppressed, and there is now a hard error when starting a DXVK game without Vulkan.

Perhaps that warning on startup can just go away? And perhaps those warnings should offer the choice to start anyway? Or maybe not; maybe users won't read them if they can click 'play anyway' to ignore them.

**Comment by strycore** [comment]
Install Vukan properly and the message dialog will go away.

**Reply by telanus** [reply]
well I don't have a Vulkan capable card, and I don't have the finances to get a gpu that is vulkan capable.

**Reply by strycore** [reply]
Ok, then we should provide a real solution, not telling the user to install Vulkan when there isn't Vulkan support to begin with.

Can you give me the PCI IDs of your GPU? It's displayed when you start `lutris -d` and looks like something like:

`INFO     2023-06-15 22:24:16,612 [startup.check_driver:85]:GPU: 10DE:2504 196E:138E (nvidia drivers)`

**Reply by telanus** [reply]
GPU: 10DE:0F00 10DE:098F (nvidia drivers)

**Reply by strycore** [reply]
Got it. I'll come with something that bypasses the error dialog.

**Comment by strycore** [comment]
Dismissable error messages are indeed gone. Either the error can be fixed, or it's not an error just a different configuration what we need to take into account.

We could show no warning if Vulkan isn't installed but disable DXVK and VKD3D options

**Comment by strycore** [comment]
Now that I have absolutely every game I could think of playable on Lutris, the goal will shift into making Lutris a LOT more strict regarding its configuration. This will only stop once I stop seeing people pop in the Discord to report issues that can't be reproduced.

**Reply by danieljohnson2** [reply]
Good grief! Nothing will ever stop people reporting unreproducible issues- some of them will even be real issues.

**Reply by strycore** [reply]
That's the whole point, close down on actual issues while letting users without actual issues use the software without warnings. In this case, it's good to make a difference between "This setup doesn't have Vulkan setup correctly" from "This hardware doesn't support Vulkan, just disable every Vulkan feature"

**Reply by danieljohnson2** [reply]
I'll take a stab at disabling Vulkan features when I get a chance, if you do not get there first.

**Reply by danieljohnson2** [reply]
OK, I've committed revised Vulkan error reporting. No more dialog on game start, instead the runner config looks like this:

![Screenshot from 2023-06-17 05-05-01](https://github.com/lutris/lutris/assets/6507403/d0988815-84c5-4125-a29d-c2a5aecf33b0)

In addition, DXVK and VKD3D will be disabled, even if the config says to use them.

(edit: revised messages to be less repetitive)

**Reply by telanus** [reply]
Hi @danieljohnson2 there is still this one that's not dealt with:
![image](https://github.com/lutris/lutris/assets/19357986/b41baf4b-57be-4f3c-810e-f8a738f77114)

This one lives in startup.py

**Reply by danieljohnson2** [reply]
I looked and found a *different* startup check- for missing per-architecture Vulkan libraries. I've moved that into my new UI also. Once again, if your libs are messed up, you won't be able to even try to use DXVK and VKD3D.

I think this needs to cover DXVK-NVAPI too, so I've disabled that too, when Vulkan is not there.

So, the outdated NVIDIA driver thing. It wants version 415; the message suggests this is where Vulkan support appears. But my research suggests otherwise- it seems to have already been there.

I am not sure I should disable DXVK for you. Can you use DXVK or VKD3D at all, @telanus? Maybe the issue is that those old drivers are just too crap to work.

@strycore, I could use some insight on this. Was this outdated-drivers error just about Vulkan? Should it mean you can't even try to use DXVK, like Vulkan was missing or broken?

**Reply by telanus** [reply]
Personally I don't use Vulkan, the closest i get to it is in [Star Trek: Encounters](https://www.mobygames.com/game/24384/star-trek-encounters/) :P 

I normally play Emulated games or on Steam, and then it's normally older games. The driver is one of the earliest vilkan drivers.

For myself, to bypass the check when starting, I just removed that part of the code

**Reply by danieljohnson2** [reply]
Seems like the driver works well enough for basic WINE. Hmm.

**Reply by strycore** [reply]
That driver check was added to support newer features of Vulkan. If Vulkan is disabled then this message can go.

**Reply by danieljohnson2** [reply]
Hmm. I guess we want the message since this long predates Vulkan 1.3, so it's not a warning we have elsewhere. But this could be a warning in the configuration window. I'll move it.

**Reply by danieljohnson2** [reply]
OK, I've moved that warning into the configuration dialog like the others. That should be the last startup warning dialog.
